src/maliput_malidrive/road_curve/spiral_ground_curve.cc

```diff
@@ -0,0 +1,206 @@
+}
+
+// @return The coefficients of the series expansion for a Fresnel sine.
+constexpr std::array<double, kNumCoefficients> GetFresnelSinCoefficients() {
+  constexpr std::array<double, kNumCoefficients> coeff{
+      PowerOf(-1., 0) / (double{Factorial(2L * 0L + 1L)} * (4. * 0. + 3.)),
+      PowerOf(-1., 1) / (double{Factorial(2L * 1L + 1L)} * (4. * 1. + 3.)),
+      PowerOf(-1., 2) / (double{Factorial(2L * 2L + 1L)} * (4. * 2. + 3.)),
+      PowerOf(-1., 3) / (double{Factorial(2L * 3L + 1L)} * (4. * 3. + 3.)),
+      PowerOf(-1., 4) / (double{Factorial(2L * 4L + 1L)} * (4. * 4. + 3.)),
+      PowerOf(-1., 5) / (double{Factorial(2L * 5L + 1L)} * (4. * 5. + 3.)),
+      PowerOf(-1., 6) / (double{Factorial(2L * 6L + 1L)} * (4. * 6. + 3.)),
+      PowerOf(-1., 7) / (double{Factorial(2L * 7L + 1L)} * (4. * 7. + 3.)),
+      PowerOf(-1., 8) / (double{Factorial(2L * 8L + 1L)} * (4. * 8. + 3.)),
+      PowerOf(-1., 9) / (double{Factorial(2L * 9L + 1L)} * (4. * 9. + 3.)),
+  };
+  return coeff;
+}
+
+// @return The exponents of series expansion for a Fresnel cosine.
+constexpr std::array<double, kNumCoefficients> GetFresnelCosExponents() {
+  constexpr std::array<double, kNumCoefficients> exponents{
+      4. * 0. + 1., 4. * 1. + 1., 4. * 2. + 1., 4. * 3. + 1., 4. * 4. + 1.,
+      4. * 5. + 1., 4. * 6. + 1., 4. * 7. + 1., 4. * 8. + 1., 4. * 9. + 1.,
+  };
+  return exponents;
+}
+
+// @return The exponents of series expansion for a Fresnel sine.
+constexpr std::array<double, kNumCoefficients> GetFresnelSinExponents() {
+  constexpr std::array<double, kNumCoefficients> exponents{
+      4. * 0. + 3., 4. * 1. + 3., 4. * 2. + 3., 4. * 3. + 3., 4. * 4. + 3.,
+      4. * 5. + 3., 4. * 6. + 3., 4. * 7. + 3., 4. * 8. + 3., 4. * 9. + 3.,
+  };
+  return exponents;
+}
+
+// Computes the position of a Fresnel spiral at @p t.
+// @param t The normalized spiral parameter.
+// @return The position of a Fresnel spiral at @p t.
+maliput::math::Vector2 FresnelSpiral(double t) {
+  static constexpr std::array<double, kNumCoefficients> cos_exponents = GetFresnelCosExponents();
+  static constexpr std::array<double, kNumCoefficients> sin_exponents = GetFresnelSinExponents();
+  static constexpr std::array<double, kNumCoefficients> cos_coefficients = GetFresnelCosCoefficients();
+  static constexpr std::array<double, kNumCoefficients> sin_coefficients = GetFresnelSinCoefficients();
+
+  std::array<double, kNumCoefficients> x_terms{};
+  std::array<double, kNumCoefficients> y_terms{};
+  std::fill(x_terms.begin(), x_terms.end(), t);
+  std::fill(y_terms.begin(), y_terms.end(), t);
+
+  for (size_t i = 0; i < kNumCoefficients; ++i) {
+    x_terms[i] = PowerOf(x_terms[i], cos_exponents[i]);
+    y_terms[i] = PowerOf(y_terms[i], sin_exponents[i]);
+  }
+
+  return maliput::math::Vector2{
+      std::inner_product(x_terms.begin(), x_terms.end(), cos_coefficients.begin(), 0.),
+      std::inner_product(y_terms.begin(), y_terms.end(), sin_coefficients.begin(), 0.),
+  };
+}
```

Comment:
Maybe we could move all fresnel related stuff to a different header file (still private) so we can test it decoupled from the SpiralGroundCurve implementation.  (maybe not worth it to be added to maliput math yet.)